File: problems-originals/2401-2500/2405_optimal-partition-of-string/solution.py
class Solution:
    def partitionString(self, s: str) -> int:
        count = 1
        seen = 0
        for ch in s:
            bit = 1 << (ord(ch) - ord("a"))
            if seen & bit:
                count += 1
                seen = bit
            else:
                seen |= bit
        return count

File: problems-originals/2401-2500/2405_optimal-partition-of-string/solution.rs
impl Solution {
    pub fn partition_string(s: String) -> i32 {
        let bytes = s.as_bytes();
        let mut count = 1;
        let mut seen: u32 = 0;
        for &b in bytes {
            let bit = 1u32 << (b - b'a');
            if seen & bit != 0 {
                count += 1;
                seen = bit;
            } else {
                seen |= bit;
            }
        }
        count
    }
}
